tests/lib/docsis.py

```diff
@@ -84,3 +96,68 @@ def encode_cm():
             return encode_cm()
         except:
             return encode_mta()
+
+class cm_cfg(object):
+    '''
+    Class for generating CM cfg from nothing, or even importing from a file
+    They later need to be encoded via a compiler
+    '''
+
+    # TODO: all these names will need to be made up once we don't have
+    # an input file anymore
+    original_fname = None
+    original_file = None
+    encoded_suffix = '.cfg'
+    encoded_fname = None
+
+    # string representation of cm cfg
+    # temporary for starting point
+    txt = ""
+
+    # plenty of tests reference a file name, and assume it's in a certain
+    # place so let's allow for that for now
+    legacy_search_path = None
+
+    def __init__(self, start=None):
+        '''Creates a default basic CM cfg file for modification'''
+
+        # TODO: we require loading a file for the moment
+        if start == None:
+            raise Exception("We don't know how to start from scratch yet")
+        else:
+            # TODO: filename should not matter
+            self.original_file = start
+            self.original_fname = os.path.split(start)[1]
```

Comment:
Nit: could this be `os.path.basename` ?

Reply:
Yes, it should... copying from input args... but for now I don't have a good test so I don't want to risk it in this commit.